string objective
string[] equality_constraints
string[] inequality_constraints
